demos/aks-nat-agic/azuredeploy.json

```diff
@@ -1402,14 +1403,14 @@
           }
         },
         {
+          "comments": "Grant the AKS cluster ingress controller pod managed identity with reader role permissions over Key Vault; paired with the Access Policy, this allows our ingress controller to pull certificates.",
           "type": "providers/roleAssignments",
           "apiVersion": "2020-04-01-preview",
-          "name": "[concat('Microsoft.Authorization/', guid(concat(resourceGroup().id), variables('readerRoleId')))]",
+          "name": "[concat('Microsoft.Authorization/', guid(concat(resourceGroup().id), variables('readerRoleId'), parameters('aksClusterName')))]",
```

Comment:
Remove the concat() from the guid() functions (That will change the seed)

"[concat('Microsoft.Authorization/', guid(resourceGroup().id), variables('readerRoleId'), parameters('aksClusterName'))]",

Reply:
Fixed